EIPS/eip-6909.md

```diff
@@ -0,0 +1,600 @@
+
+### Removal of Required Callbacks
+
+Callbacks MAY be used within a multi-token compliant contract, but it is not required. This allows for more gas efficient methods by reducing external calls and additional checks.
+
+### Removal of "Safe" Naming
+
+The `safeTransfer` and `safeTransferFrom` naming conventions are misleading, especially in the context of the ERC-1155 and [ERC-721](./eip-721.md) standards, as they require external calls to receiver accounts with code, passing the execution flow to an arbitrary contract, provided the receiver contract returns a specific value. The combination of removing mandatory callbacks and removing the word "safe" from all method names improves the safety of the control flow by default.
+
+### Interface ID
+
+The interface ID is `0xb2e69f8a`.
+
+### Metadata Extension
+
+#### Methods
+
+##### name
+
+The `name` of the contract.
+
+```yaml
+- name: name
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: name
+      type: string
+```
+
+##### symbol
+
+The ticker `symbol` of the contract.
+
+```yaml
+- name: symbol
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: symbol
+      type: string
+```
+
+##### decimals
+
+The `amount` of decimals for a token `id`.
+
+```yaml
+- name: decimals
+  type: function
+  stateMutability: view
+
+  inputs:
+    - name: id
+      type: uint256
+
+  outputs:
+  - name: amount
+    type: uint8
+```
+
+### Metadata URI Extension
+
+#### Methods
+
+##### tokenURI
+
+The `URI` for a token `id`.
+
+MAY revert if the token `id` does not exist.
+
+MUST replace occurrences of `{id}` in the returned URI string by the client.
+
+```yaml
+- name: tokenURI
+  type: function
+  stateMutability: view
+
+  inputs:
+    - name: id
+      type: uint256
+
+  outputs:
+    - name: uri
+      type: string
+```
+
+#### Metadata Structure
+
+The metadata specification closely follows that of the ERC-721 JSON schema.
+
+MUST replace occurrences of `{id}` in the returned URI string by the client.
+
+```json
+{
+  "title": "Asset Metadata",
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string",
+      "description": "Identifies the token"
+    },
+    "description": {
+      "type": "string",
+      "description": "Describes the token"
+    },
+    "image": {
+      "type": "string",
+      "description": "A URI pointing to an image resource."
+    }
+  }
+}
+```
+
+## Backwards Compatibility
+
+This is not backwards compatible with ERC-1155 as some methods are removed. however, wrappers can be implemented for the ERC-20, ERC-721, and ERC-1155 standards.
+
+## Reference Implementation
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code inline in the EIP must be `CC0-1.0` licensed. Code in the assets directory can be MIT.